public/app/features/expressions/ExpressionDatasource.ts

```diff
@@ -61,10 +61,9 @@ export class ExpressionDatasourceApi extends DataSourceWithBackend<ExpressionQue
 }
 
 /**
- * MATCHES a constant in DataSourceWithBackend, this should be '__expr__'
- * @deprecated
+ * MATCHES a constant in DataSourceWithBackend
  */
-export const ExpressionDatasourceUID = '-100';
+export const ExpressionDatasourceUID = '__expr__';
```

Comment:
I am not sure if that is the ultimate solution to the problem but that definitely solves it. Please advise if there is something better or wiser we can do.